### PR TITLE
test: v1 to v2 graphql migration test package

### PR DESCRIPTION
**Pull request by edwardfoyle**
#### Description of changes

Creating a test framework for migration from v1 to v2 of the graphql transformer. The tests check that certain invariants hold true of the CFN output from both the v1 transformer and the v2 transformer. The tests internally run the schema migration script to update the gql schema from v1 to v2.

In their current state, the tests will assert that DynamoDB tables are not removed / replaced during migration. We can add additional assertions on the stack diffs moving forward.

#### Description of how you validated changes
Ran the test suite locally against a sample schema and hand-built migration schema
#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-graphql-migration-tests/src/v2-transformer-provider.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    indexTransformer,
+    new BelongsToTransformer(),
+    new HasManyTransformer(),
+    hasOneTransformer,
+    new ManyToManyTransformer(modelTransformer, indexTransformer, hasOneTransformer, authTransformer),
+    new DefaultValueTransformer(),
+    authTransformer,
+  ];
+
+  // if (options?.addSearchableTransformer) {
```
Comment (cjihrig) [inline packages/amplify-graphql-migration-tests/src/v2-transformer-provider.ts:62]: Is this and the predictions transformer above supposed to be commented out?

File: packages/amplify-graphql-migration-tests/src/v1-transformer-provider.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+const defaultConfig: V1TransformerTestConfig = {
+  transformers: [
+    new DynamoDBModelTransformer(),
+    new VersionedModelTransformer(),
+    // new FunctionTransformer(),
+    new HttpTransformer(),
+    new KeyTransformer(),
+    new ModelConnectionTransformer(),
+    // new SearchableModelTransformer(),
```
Comment (cjihrig) [inline packages/amplify-graphql-migration-tests/src/v1-transformer-provider.ts:20]: Same question here about searchable and function being commented out.

File: packages/amplify-graphql-migration-tests/src/test-case-registry.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+/**
+ * REGISTER TEST CASES HERE
+ */
+export const getTestCaseRegistry = (): TestEntry[] => [
+  {
+    name: 'basic bi-di has-one and has-many connection',
+    schema: basicBiDiRelationSchema,
+  },
+  /*
```
Comment (cjihrig) [inline packages/amplify-graphql-migration-tests/src/test-case-registry.ts:11]: Should this be commented out?

**Review by cjihrig — COMMENTED** [review]
Left a few comments - mostly just about commented code. There are also some LGTM bot issues.

File: packages/amplify-graphql-migration-tests/src/v2-transformer-provider.ts (hunk 1 above)
Comment (edwardfoyle) [reply packages/amplify-graphql-migration-tests/src/v2-transformer-provider.ts:62]: removed these for now. We can add them back if / when we test schemas that use these directives

File: packages/amplify-graphql-migration-tests/src/test-case-registry.ts (hunk 3 above)
Comment (edwardfoyle) [reply packages/amplify-graphql-migration-tests/src/test-case-registry.ts:11]: yeah this is meant to be a comment showing how to add more schemas to the test registry

**Review by cjihrig — APPROVED** [review]

**Pull request merged**